### DISCUSSION TITLE
HTML self closing tags (eg img) is not formatting correctly.

### DISCUSSION THREAD

**Discussion by featuriz** [apache/netbeans#7336] [General]
Better seen in the screenshot. How bad it is. Even for a simple HTML tags.
![Screenshot from 2024-04-26 12-49-35](https://github.com/apache/netbeans/assets/10089154/324c6291-5a38-41ec-9b1c-6e74a3049a44)

**Comment by matthiasblaesing** [comment]
@featuriz feel free to dive into the HTML infrastructure of NetBeans. If I remember correctly NetBeans uses an HTML parser to do the formatting, so in theory it should be possible to handle empty elements. The problem might be the SVG element in the middle, which might or might not be a problem for the parser.

Side rant: This problem was created more than 10 years ago when the good way of serializing HTML as an XML language was abandoned and the current hell was created. All because some script kiddies failed to write simplest syntax.

**Reply by featuriz** [reply]
Thanks for your time. Yes you are right it was working fine under sun and oracle. (Just Kidding).  
**Note:** I found in other html files and templating file like twig. The error is same. `<img src="" alt="" >` => these types of self or no closing tags makes problem. If I add  `<img src="" alt="" />`  or  `<img src="" alt="" ></img>` , then it formats correctly. 
**Problem:** I can't update all codes from a large project. (It's not in my hand/radar).

END: It should know self closing tags and have to properly format it.

**Reply by matthiasblaesing** [reply]
This looks right from a formatting perspective:

![image](https://github.com/apache/netbeans/assets/2179736/ec782a21-2186-4852-a965-55ab71e91a28)

The `path` elements looks off, but the sequence of `img` and `div` elements looks good. Please provide a minimal sample project that shows the problem.

**Reply by Chris2011** [reply]
For me, this is also working just fine, except from the path tag. I thought also it was of the missinterpreting HTML5 tags which can be self closed or not, doesnt matter which tag anymore but netbeans can handle this:

```html
<section>
  <div class="gap—8 items—center py—8 px—4 mx—auto">
    <img class="dark:hidden" src="">
    <img class="smt"
         src="#" alt="dashboard image">
    <div>
      <h2 class="mb—4 text—4x1 tracking tont—light text—gr"></h2>
      <a class="inline—flex it Get started"></a>
      <svg class="m1—2 —mr—l w—5 h—">
      <path />
      </svg>
    </div>
  </div>
</section>
```

it was just extracted with OCR so I updated the HTML for myself.

**Reply by matthiasblaesing** [reply]
I followed my gut feeling and tested another setup. I created ``scratch.xhtml` and pasted the code from @Chris2011. With that I get the broken formatting. But then NetBeans is still right, as `xhtml` is the XML serialization of HTML and thus `<img ...>` is not self-closing. @featuriz as you did not provide a testable project, I have to conclude, that you provided an invalid example.

**Reply by featuriz** [reply]
Sorry to say this. I completely uninstalled NetbeansIDE from my system. Now I'm working on VSCODE. 
By the way, "Thanks to Netbeans problem. I found a better editor"

**Reply by Chris2011** [reply]
Sry to hear that but as we said we needed more infos to fix this problem if this is one at our side. But until now we didn't get the info that we asked for. We would like to make it better and also fix the problem if there is one.
